Mage.Sets/src/mage/cards/c/ChromaticOrrery.java

```diff
@@ -0,0 +1,84 @@
+package mage.cards.c;
+
+import mage.Mana;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.AsThoughManaEffect;
+import mage.abilities.effects.common.DrawCardForEachColorAmongControlledPermanentsEffect;
+import mage.abilities.mana.SimpleManaAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.players.ManaPoolItem;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class ChromaticOrrery extends CardImpl {
+
+    public ChromaticOrrery(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{7}");
+        
+        this.addSuperType(SuperType.LEGENDARY);
+
+        // You may spend mana as though it were mana of any color.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new ChromaticOrreryEffect()));
+
+        // {T}: Add {C}{C}{C}{C}{C}.
+        this.addAbility(new SimpleManaAbility(Zone.BATTLEFIELD, Mana.ColorlessMana(5), new TapSourceCost()));
+
+        // {5}, {T}: Draw a card for each color among permanents you control.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new DrawCardForEachColorAmongControlledPermanentsEffect(), new GenericManaCost(5));
```

Comment:
This isn't important but you can leave out the zone since the default constructor already assumes it

Reply:
updated